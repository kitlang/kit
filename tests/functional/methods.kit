struct MyStruct {
    public var a: Int16;

    public function myMethod() {
        printf("a = %i\n", this.a);
    }
}

function staticExtension(m: MyStruct, i: Int) {
    printf("static extension called with %i and %i\n", m.a, i);
}

function main() {
    var s1 = struct MyStruct {a: 1};
    s1.myMethod();
    var s2 = struct MyStruct {a: 2};
    MyStruct.myMethod(s2);
    var s3 = struct MyStruct {a: 3};
    using implicit s3 {
        MyStruct.myMethod();
    }

    staticExtension(s1, 5);
    s2.staticExtension(6);
}
