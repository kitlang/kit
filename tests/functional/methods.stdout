a = 1
a = 2
a = 3
static extension called with 1 and 5
static extension called with 2 and 6
